R/spark_gateway.R

```diff
@@ -57,7 +57,7 @@ spark_gateway_connection <- function(master, config, gatewayInfo, gatewayAddress
   })
 
   # create the shell connection
-  sc <- structure(class = c("spark_connection", "spark_gateway_connection", "spark_shell_connection"), list(
```

Comment:
This change broke Databricks connection. Our test against master failed with:

~~~
Error in value[[3L]](cond) : 
  Failed during initialize_connection:Failed during initialize_connection: Unable to retrieve a spark_connection from object of class NULL
~~~

The error is from `spark_connection.default`. With the new refactor, we might need an implementation of `spark_connection` for Databricks.

@falaki @javierluraschi @kevinykuo Any recommendation for the fix?